system/loggerd/uploader.py

```diff
@@ -152,14 +151,12 @@ def do_upload(self, key: str, fn: str):
       return FakeResponse()
 
     with open(fn, "rb") as f:
-      data: BinaryIO
       if key.endswith('.zst') and not fn.endswith('.zst'):
         compressed = zstd.compress(f.read(), LOG_COMPRESSION_LEVEL)
-        data = io.BytesIO(compressed)
+        with io.BytesIO(compressed) as data:
+          return requests.put(url, data=data, headers=headers, timeout=10)
       else:
-        data = f
-
-      return requests.put(url, data=data, headers=headers, timeout=10)
+        return requests.put(url, data=f, headers=headers, timeout=10)
```

Comment:
Can you please copy the style you chose in https://github.com/commaai/openpilot/pull/30237?